pkg/executor/executortype/poolmgr/gp_deployment.go

```diff
@@ -62,6 +62,14 @@ func (gp *GenericPool) genDeploymentMeta(env *fv1.Environment) metav1.ObjectMeta
 		Name:        getPoolName(env),
 		Labels:      deployLabels,
 		Annotations: deployAnnotations,
+		OwnerReferences: []metav1.OwnerReference{
```

Comment:
This change would be applicable for other executor types and other kind of resources we create through fission as well.
We need to see what all places we can set owner references and set them accordingly.

Reply:
Yes, might also be added to buildmgr deployments if adding this to all. 

Btw, thanks for the work! 🫶🏻

Reply:
Set OwnerReferences to following K8s resources created by Fission:
- Set OwnerReferences to deployment, service and HPA created by newdeploy executortype.
- Set OwnerReferences to deployment, service and HPA created by container function.
- Set OwnerReferences to builderManager deployment and service created by Fission's environment resource.